newsrevenuehub/rev-engine
# DEV-1992 Flake8 test code and settings

#### What's this PR do?
Apply flake8 linter to test code

  - Change .pre-commit-config.yaml Flake8 config to not ignore tests.
  - Change .pre-commit-config.yaml Flake8 and .flake8 to have matching excludes.
  - Add ignore E731 "Do not assign a lambda expression". Too opinionated.
  - Cut ignore F841 unused variable.
  - Cut unused imports and variables from test code.
  - Cut three tests with duplicate name. Cut later ones, the ones not run.
    One was literal duplicate, the other two were smaller and seemed older.

#### Why are we doing this? How does it help us?
While it could be argued test code needn't follow as stringent linting
as production code, Flake8 also catches errors such as duplicate
var/method/function names. So we should still lint test code.

#### How should this be manually tested?
checkout code base, run flake8 locally and how pre-commit/CI would run it. ensure no reported issues

> flake8
> pre-commit run flake8 --all

Then add flake8 violation (e.g. unused import), run again and ensure flake8 reports issue.
 
#### Have automated unit tests been added?
N/A "yes" sort of the pre-commit runs this.

#### How should this change be communicated to end users?
N/A

#### Are there any smells or added technical debt to note?
The settings and migrations are not flake8. 

#### Has this been documented? If so, where?
N/A

#### What are the relevant tickets? Add a link to any relevant ones.
[DEV-1992]

#### Do any changes need to be made before deployment to production (adding environment variables, for example)?
No.

#### Are there next steps? If so, what? Have tickets been opened for them?
No.

[DEV-1992]: https://news-revenue-hub.atlassian.net/browse/DEV-1992?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
